### PR TITLE
API: `optimize.check_grad`: transition to Generator (SPEC 7)

### PR DESCRIPTION
**Pull request by andyfaff**

### CODE REVIEW

File: scipy/optimize/_optimize.py [hunk 1]
```diff
@@ -1056,14 +1057,31 @@ def check_grad(func, grad, x0, *args, epsilon=_epsilon,
...
         If `func` is a vector valued function then only ``'all'`` can be used.
-    seed : {None, int, `numpy.random.Generator`, `numpy.random.RandomState`}, optional
-        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
-        singleton is used.
-        If `seed` is an int, a new ``RandomState`` instance is used,
-        seeded with `seed`.
-        If `seed` is already a ``Generator`` or ``RandomState`` instance then
-        that instance is used.
-        Specify `seed` for reproducing the return value from this function.
+    rng : {None, int, `numpy.random.Generator`, `numpy.random.RandomState`}, optional
```
Comment (andyfaff) [inline scipy/optimize/_optimize.py:1060]: I didn't use `replace_doc` in the decorator because there's extra information at the bottom of the docstring.

Comment (mdhaber) [reply scipy/optimize/_optimize.py]: Maybe I'll add a rule to the decorator to replace only the paragraph before a blank line?

Comment (andyfaff) [reply scipy/optimize/_optimize.py]: I think that makes sense

**Review by mdhaber — APPROVED** [review]
LGTM. Can merge when CI is happy.

**Pull request merged**